types/src/system/handle_payment/mod.rs

```diff
@@ -118,10 +118,24 @@ mod internal {
         }
 
         // User's part
-        let refund_amount = (total - amount_spent) * REFUND_PERCENTAGE;
+        let refund_amount = {
+            let refund_amount_raw = match total.checked_sub(amount_spent) {
```

Comment:
Three more cases where we should use `ok_or()` for brevity in this file.

Reply:
Applied suggestion